### PR TITLE
fix: make v10 calls exclusive to v10 procedures

### PR DESCRIPTION
**Pull request by juliusmarminge**
showcase prisma starter with a new procedure from following migration guide

### CODE REVIEW

File: examples/.interop/next-prisma-starter-migration/src/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  const addPost = trpc.useMutation('post.add', {
+    async onSuccess() {
+      // refetches posts after a post is added
+      await utils.invalidateQueries(['post.all']);
+    },
+  });
+
+  // @ts-expect-error - should not be possible to call old procedures using proxy
+  const { data: yay } = proxy.healthz.useQuery();
+  //            ^?
```
Comment (juliusmarminge) [inline examples/.interop/next-prisma-starter-migration/src/pages/index.tsx:17]: this should error but it doesn't.

no need to make the proxy object include old procs?

File: examples/.interop/next-prisma-starter-migration/src/server/routers/_app.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  })
+  /**
+   * Merge `postRouter` under `post.`
+   */
+  .merge('post.', postRouter)
+  .interop();
+
+const newRouter = t.router({
+  newQuery: t.procedure.query(() => 'Hello v10'),
+});
```
Comment (juliusmarminge) [inline examples/.interop/next-prisma-starter-migration/src/server/routers/_app.ts:42]: this should be the only thing callable using proxy

File: examples/.interop/next-prisma-starter-migration/.env [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Make sure to override these in deployment
```
Comment (mshd) [inline examples/.interop/next-prisma-starter-migration/.env:1]: env should not be in git by default, maybe rename it to .env.example

Comment (juliusmarminge) [reply examples/.interop/next-prisma-starter-migration/.env:1]: Thanks for spotting.

Just threw this up as showcase for an internal discussion, probably more to do before it's ready.

**Pull request merged**